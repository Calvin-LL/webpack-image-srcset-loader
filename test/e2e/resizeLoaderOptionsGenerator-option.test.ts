import WISLWebpackTestCompiler from "./helpers/WISLWebpackTestCompiler";

describe.each([4, 5] as const)(
  'v%d "resizeLoaderOptionsGenerator" option',
  (webpackVersion) => {
    it("should work with a function", async () => {
      const mockResizeLoaderOptionsGenerator: (
        width: number | undefined,
        scale: number | undefined,
        existingOptions: Record<string, any>
      ) => Record<string, any> = jest.fn().mockReturnValue({
        fileLoaderOptions: {
          esModule: false,
        },
      });

      const compiler = new WISLWebpackTestCompiler({ webpackVersion });
      const bundle = await compiler.compile({
        loaderOptions: {
          sizes: ["2x", "original", "1x", "300w"],
          resizeLoaderOptionsGenerator: (
            width: number | undefined,
            scale: number | undefined,
            existingOptions: Record<string, any>
          ) => mockResizeLoaderOptionsGenerator(width, scale, existingOptions),
        },
      });

      expect(bundle.execute("main.js")).toMatchSnapshot("result");

      expect(mockResizeLoaderOptionsGenerator).toHaveBeenCalled();
      expect(mockResizeLoaderOptionsGenerator).toMatchSnapshot();
    });
  }
);
