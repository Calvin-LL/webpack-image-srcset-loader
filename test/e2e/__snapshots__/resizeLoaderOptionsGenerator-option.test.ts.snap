// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v4 "resizeLoaderOptionsGenerator" option should work with a function 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      undefined,
      1,
      undefined,
    ],
    Array [
      undefined,
      undefined,
      undefined,
    ],
    Array [
      undefined,
      0.5,
      undefined,
    ],
    Array [
      300,
      undefined,
      undefined,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
  ],
}
`;

exports[`v4 "resizeLoaderOptionsGenerator" option should work with a function: result 1`] = `"bd2b9fdf768b5eb139702c4c3b4573de.jpeg 2x, bd2b9fdf768b5eb139702c4c3b4573de.jpeg, bd2b9fdf768b5eb139702c4c3b4573de.jpeg 1x, bd2b9fdf768b5eb139702c4c3b4573de.jpeg 300w"`;

exports[`v5 "resizeLoaderOptionsGenerator" option should work with a function 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      undefined,
      1,
      undefined,
    ],
    Array [
      undefined,
      undefined,
      undefined,
    ],
    Array [
      undefined,
      0.5,
      undefined,
    ],
    Array [
      300,
      undefined,
      undefined,
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
    Object {
      "type": "return",
      "value": Object {
        "fileLoaderOptions": Object {
          "esModule": false,
        },
      },
    },
  ],
}
`;

exports[`v5 "resizeLoaderOptionsGenerator" option should work with a function: result 1`] = `"bd2b9fdf768b5eb139702c4c3b4573de.jpeg 2x, bd2b9fdf768b5eb139702c4c3b4573de.jpeg, bd2b9fdf768b5eb139702c4c3b4573de.jpeg 1x, bd2b9fdf768b5eb139702c4c3b4573de.jpeg 300w"`;
